### PR TITLE
Extends tests for xemu #950.

### PR DESCRIPTION
**Pull request by abaire**
Adds corner cases to ensure that the blit overlap detection in https://github.com/xemu-project/xemu/pull/2211 is correct

### CODE REVIEW

File: src/tests/image_blit_tests.cpp [hunk 1]
```diff
@@ -592,6 +649,92 @@ void ImageBlitTests::TestDirtyOverlappedDestinationSurface() {
...
+
+  constexpr uint32_t kSurfW = 128;
+  constexpr uint32_t kSurfH = 128;
+  constexpr uint32_t kSurfX = 64;
+  constexpr uint32_t kSurfY = 64;
+  const uint32_t kBackbufferPitch = host_.GetFramebufferWidth() * 4;
+
+  auto subtexture = reinterpret_cast<uint8_t*>(pb_back_buffer()) + kSurfY * kBackbufferPitch + kSurfX * 4;
+
+  host_.RenderToSurfaceStart(subtexture, TestHost::SCF_A8R8G8B8, host_.GetFramebufferWidth(), kSurfH);
```
[the remark this answers is not in the document]
Comment (abaire) [reply src/tests/image_blit_tests.cpp:671]: This is intentional.
